src/sdk/namespace_merge.js

```diff
@@ -25,9 +25,18 @@ class NamespaceMerge {
     // OBJECT LIST //
     /////////////////
 
-    list_objects(params, object_sdk) {
-        return this._ns_map(ns => ns.list_objects(params, object_sdk))
-            .then(res => this._handle_list(res, params));
+    async list_objects(params, object_sdk) {
+        let res = (await this._ns_map(async function(ns) {
+            try {
+                let ns_objects = await ns.list_objects(params, object_sdk);
+                return ns_objects;
+            } catch (err) {
+                if (err.code === 'NoSuchBucket') return;
+                throw err;
+            }
+        })).filter(objects => objects);
+        if (!res || !res.length) throw new S3Error(S3Error.NoSuchBucket);
```

Comment:
What about buckets with no objects?

Reply:
great question - the call to list_objects() function on a bucket that doesn't contain any objects returns: { objects: [], common_prefixes: [], is_truncated: false, next_marker: undefined } and not just undefined..

Reply:
Gotcha!